benchmark/src/util.ts

```diff
@@ -0,0 +1,107 @@
+import { BenchmarkFunction } from "./benchmark_types";
+
+export type Result<T, E> = Ok<T, E> | Err<T, E>;
+
+export class Ok<T, E> {
```

Comment:
This seems a little heavy and awkward, what I usually do is
```ts
type Result<T> = { success: true, value: T } | { success: false, error: string };
```
And then use TypeScript's type system to do the type narrowing
```ts
if (result.success) {
    // result.value available
} else {
    // result.error available
}

Reply:
IMO monads in non-functional languages look kind of weird in general. Why not just throw a regular error? It's effect of jumping through the stack is useful in the majority of cases, especially there, where the only thing we could do on error is die.